reana_server/fetcher.py

```diff
@@ -18,20 +18,43 @@
 
 from git import Repo
 
-from reana_server.config import WORKFLOW_SPEC_EXTENSIONS, WORKFLOW_SPEC_FILENAMES
+from reana_server.config import (
+    REGEX_CHARS_TO_REPLACE,
+    WORKFLOW_SPEC_EXTENSIONS,
+    WORKFLOW_SPEC_FILENAMES,
+)
+
+
+class ParsedUrl:
+    """Utility class to parse and get information about a given URL."""
+
+    def __init__(self, url: str):
+        """Initialize the ParsedUrl class.
+
+        :param url: URL to be parsed.
+        """
+        self.original_url = url
+        self._parsed_url = urlparse(url)
+        self.path = self._parsed_url.path.rstrip("/")
+        self.dirname, self.basename = os.path.split(self.path)
+        self.stem, self.extension = os.path.splitext(self.basename)
```

Comment:
Stem is the name of a file with the extension removed (see for example the [stem function](https://en.cppreference.com/w/cpp/filesystem/path/stem) of C++ or the [Wikipedia section about filename extensions](https://en.wikipedia.org/wiki/Filename#Filename_extensions)), but I agree that it might not be clear at first sight. Do you have any naming suggestions? We might otherwise use `filename_without_ext`, what do you think?

Reply:
`stem` is actually a valid word. I learned it yesterday. 

But still would be good to make it as dump as possible. I think, `filename_without_extension` should be good.